beacon_node/lighthouse_network/src/peer_manager/peerdb.rs

```diff
@@ -673,17 +676,62 @@ impl<E: EthSpec> PeerDB<E> {
     }
 
     /// Updates the connection state. MUST ONLY BE USED IN TESTS.
-    pub fn __add_connected_peer_testing_only(&mut self, peer_id: &PeerId) -> Option<BanOperation> {
+    pub fn __add_connected_peer_testing_only(
+        &mut self,
+        supernode: bool,
+        spec: &ChainSpec,
+    ) -> PeerId {
         let enr_key = CombinedKey::generate_secp256k1();
-        let enr = Enr::builder().build(&enr_key).unwrap();
+        let mut enr = Enr::builder().build(&enr_key).unwrap();
+        let peer_id = enr.peer_id();
+        let node_id = enr.node_id().raw().into();
+
+        let custody_subnet_count = if supernode {
+            spec.data_column_sidecar_subnet_count
+        } else {
+            spec.custody_requirement
+        };
+
+        enr.insert(
+            PEERDAS_CUSTODY_SUBNET_COUNT_ENR_KEY,
+            &custody_subnet_count,
+            &enr_key,
+        )
+        .expect("u64 can be encoded");
+
         self.update_connection_state(
-            peer_id,
+            &peer_id,
             NewConnectionState::Connected {
                 enr: Some(enr),
                 seen_address: Multiaddr::empty(),
                 direction: ConnectionDirection::Outgoing,
             },
-        )
+        );
+        let peer = self.peers.get_mut(&peer_id).expect("peer exists");
+
+        // Need to insert an empty metadata to pass the condition `on_subnet_metadata`
+        peer.set_meta_data(MetaData::V2(MetaDataV2 {
+            seq_number: 0,
+            attnets: <_>::default(),
+            syncnets: <_>::default(),
+        }));
+
+        for subnet in
+            DataColumnSubnetId::compute_custody_columns::<E>(node_id, custody_subnet_count, spec)
+        {
+            // Need to pass the tests on PeerInfo
+            // - on_subnet_metadata: no action, `Subnet::DataColumn` returns true
+            // - on_subnet_gossipsub: subnets field contains subnet
+            // - is_good_gossipsub_peer: score >= 0, which equals default score
+            // peer.set_meta_data(crate::rpc::MetaData::V2(crate::rpc::MetaDataV2 {
+            //     seq_number: 0,
+            //     attnets: <_>::default(),
+            //     syncnets: <_>::default(),
+            // }));
+            peer.insert_subnet(Subnet::DataColumn(subnet.into()));
+        }
+
+        peer_id
```

Comment:
Is all this change required? Why not just take the latest change from `das` branch? The change in #6218 fixes the custody lookup tests.